src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -795,14 +835,402 @@ def __try_add_error(error_list, detail):
+                                __get_current_complete_status_errors()
+                                __recompose_truncated_msg()
+                                    __recompose_substatus_errors()
+                                    __create_installation_tombstone
+                                    __recreate_installation_summary_json()
+                                    
+                            current_package_list_size -= diff
+                            if current_package_list_size <= max_package_list_capacity:
+                                break
+                            __get_new_size_in_bytes_after_truncation() 
+                    
+                write_with_retry_using_temp_file()  
+        """
+        status_file_size_in_bytes = len(complete_status_file_payload.encode("utf-8"))   # calc complete_status_file_payload byte size on disk
+        truncated_status_file = json.loads(complete_status_file_payload)    # reload payload into python object
+
+        if status_file_size_in_bytes > self.__internal_file_capacity:   # 126kb
+            self.composite_logger.log_debug("Begin package list truncation")
+            low_pri_index = None
+            packages_capacity_diff = 0
+            assessment_substatus_index = self.__get_substatus_index(Constants.PATCH_ASSESSMENT_SUMMARY, truncated_status_file['status']['substatus'])
+            installation_substatus_index = self.__get_substatus_index(Constants.PATCH_INSTALLATION_SUMMARY, truncated_status_file['status']['substatus'])
+
+            if assessment_substatus_index is not None:        # if assessment data exists
+                self.__assessment_substatus_msg_copy = self.__get_substatus_msg_at_index(truncated_status_file, assessment_substatus_index)
+                self.__assessment_packages_copy = self.__assessment_substatus_msg_copy['patches']
+
+            if installation_substatus_index is not None:      # if installation data exists
+                low_pri_index = self.__get_installation_low_pri_index(self.__installation_packages_copy, low_pri_index)
+                self.__installation_substatus_msg_copy = self.__get_substatus_msg_at_index(truncated_status_file, installation_substatus_index)
+                self.__installation_packages_copy = self.__installation_substatus_msg_copy['patches']
+
+            status_file_without_package_list_size = self.__get_size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # deepcopy fully copy the object avoid reference modification
+            max_package_list_capacity = self.__internal_file_capacity - status_file_without_package_list_size
+            current_package_list_size = self.__get_current_packages_size_in_bytes(self.__assessment_packages_copy, self.__installation_packages_copy)
+
+            while status_file_size_in_bytes > self.__internal_file_capacity and current_package_list_size > max_package_list_capacity:
```

Comment:
We don't need status_file_size_in_bytes > self.__internal_file_capacity, since:
1. current_package_list_size is always recomputed
2. We have a fixed value for max_package_list_capacity
So as long as current_package_list_size > max_package_list_capacity, this loop will continue

Reply:
logic is removed

Reply:
@feng-j678 the current_package_list_size is computed from status_file_size_in_bytes. So ensuring current_package_list_size is within the expected limit, also ensures status_file_size is being honored.